modules/ml/src/main/java/org/apache/ignite/ml/tree/data/DecisionTreeDataWithIndex.java

```diff
@@ -19,18 +19,14 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import org.apache.ignite.ml.dataset.primitive.FeatureMatrixWithLabelsOnHeapData;
 import org.apache.ignite.ml.tree.TreeFilter;
 
 /**
- * A partition {@code data} of the containing matrix of features and vector of labels stored in heap.
+ * A partition {@code data} of the containing matrix of features and vector of labels stored in heap
+ * with index on features.
  */
-public class DecisionTreeData implements AutoCloseable {
-    /** Matrix with features. */
-    private final double[][] features;
-
-    /** Vector with labels. */
-    private final double[] labels;
-
+public class DecisionTreeDataWithIndex extends FeatureMatrixWithLabelsOnHeapData implements AutoCloseable {
```

Comment:
Will we have in database DTDataWihoutIndex? Should it be renamed?

Reply:
Yes, you're right